api/achievements.go

```diff
@@ -56,27 +63,26 @@ func UpdateAchievement(c *gin.Context) {
 	var achievement Achievements
 	db.First(&achievement, id)
 
-	if achievement.Achievement != "" {
-		if achievement.ID != 0 {
-			var newAchievement Achievements
-			c.Bind(&newAchievement)
-
+	if achievement.ID != 0 {
+		var newAchievement Achievements
+		c.Bind(&newAchievement)
+		if newAchievement.Achievement != "" {
 			result := Achievements{
 				ID:          achievement.ID,
 				Achievement: newAchievement.Achievement,
 			}
-
 			db.Save(&result)
 			c.JSON(200, gin.H{"success": result})
 		} else {
-			c.JSON(404, gin.H{"error": "Achievement not found"})
+			c.JSON(422, gin.H{"error": "One or more of the fields are empty"})
```

Comment:
Should be along the lines of 'Achievement is missing', since there is only really one field that can be missing.